CRM/Admin/Form/ScheduleReminders.php

```diff
@@ -43,64 +43,64 @@ class CRM_Admin_Form_ScheduleReminders extends CRM_Admin_Form {
 
   public $_freqUnits;
 
+  /**
+   * Context of the form being loaded.
+   *
+   * 'event' or null
+   *
+   * @var string
+   */
+  protected $context;
+
   /**
    * Build the form object.
    */
   public function buildQuickForm() {
     parent::buildQuickForm();
     $this->_mappingID = $mappingID = NULL;
     $providersCount = CRM_SMS_BAO_Provider::activeProviderCount();
-    $this->_context = CRM_Utils_Request::retrieve('context', 'String', $this);
+    $this->context = CRM_Utils_Request::retrieve('context', 'String', $this);
+    if ($this->context == 'event') {
+      $this->_compId = CRM_Utils_Request::retrieve('compId', 'Integer', $this);
+    }
 
     //CRM-16777: Don't provide access to administer schedule reminder page, with user that does not have 'administer CiviCRM' permission
-    if (empty($this->_context) && !CRM_Core_Permission::check('administer CiviCRM')) {
-      CRM_Core_Error::fatal(ts('You do not have permission to access this page.'));
+    if (!$this->context && !CRM_Core_Permission::check('administer CiviCRM')) {
+      throw new CRM_Core_Exception(ts('You do not have permission to access this page.'));
     }
     //CRM-16777: When user have ACLs 'edit' permission for specific event, do not give access to add, delete & updtae
     //schedule reminder for other events.
-    else {
-      $this->_compId = CRM_Utils_Request::retrieve('compId', 'Integer', $this);
-      if (!CRM_Event_BAO_Event::checkPermission($this->_compId, CRM_Core_Permission::EDIT)) {
-        CRM_Core_Error::fatal(ts('You do not have permission to access this page.'));
-      }
+    elseif ($this->context === 'event' && !CRM_Event_BAO_Event::checkPermission($this->_compId, CRM_Core_Permission::EDIT)) {
```

Comment:
the actual change in this PR is the elseif vs the else

Reply:
Reading this block of code it still looks a little off. The updated version seems to say:

1. If we are in event context, you must have permission to edit this event.
2. If we have no context, you must be an administrator.
3. In any other context, you don't need any permissions at all! Happy hacking :)

Reply:
Right - I think event is the only valid context but someone could pass an invalid one - I have pushed a new version up